.github/workflows/docker.yml

```diff
@@ -38,33 +58,45 @@ jobs:
             type=raw,value=edge-${{ matrix.stage }}
             type=raw,value=latest-${{ matrix.stage }}
 
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-
-      - name: Login to DockerHub
-        if: github.event_name != 'pull_request'
-        uses: docker/login-action@v2
+      - name: Build and push Alpine images
+        uses: docker/build-push-action@v3
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          context: .
+          file: build/docker/Dockerfile
+          target: ${{ matrix.stage }}
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta-alpine.outputs.tags }}
+          labels: ${{ steps.meta-alpine.outputs.labels }}
 
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@v2
+      - name: Docker meta
+        id: meta-debian
+        uses: docker/metadata-action@v4
```

Comment:
There is a lot of duplicated configuration here. It would be nice if you could set up a matrix strategy instead: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

Reply:
thanks, I was trying to search a way of doing that without repeating myself, will try that out